x-pack/plugins/apm/server/routes/service_map/route.ts

```diff
@@ -125,10 +125,7 @@ const serviceMapRoute = createApmServerRoute({
       uiSettingsClient.get<number>(apmServiceGroupMaxNumberOfServices),
     ]);
 
-    const serviceNames = [
-      ...(serviceName ? [serviceName] : []),
-      ...(serviceGroup?.serviceNames ?? []),
-    ];
+    const serviceNames = [...(serviceName ? [serviceName] : [])];
```

Comment:
Do you still need this to be an array?

If so, I think this would be more clear:
```suggestion
    const serviceNames = compact([serviceName])
```

Reply:
I don't think this needs to be an array now cc @ogupte